helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -338,14 +525,37 @@ public Object removeRoutingTableChangeListener(
     return getInstancesForResource(resourceName, state);
   }
 
+  public Set<InstanceConfig> getInstances(String resourceName, String state,
```

Comment:
Same here. If not in the design, let's hold this for now.

Reply:
removed.

Reply:
Removed this method and other newly added public methods based in propertyType and stateType